gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java

```diff
@@ -277,28 +295,48 @@ private void addMapPin(MapPin mapPin) {
   }
 
   private void addMapPolyline(MapPolygon mapPolygon) {
-    for (ImmutableSet<Point> vertices : mapPolygon.getVertices()) {
-      PolylineOptions options = new PolylineOptions();
-
-      // Read-only
-      options.clickable(false);
-
-      // Add vertices to PolylineOptions
-      stream(vertices).map(GoogleMapsMapAdapter::toLatLng).forEach(options::add);
-
-      // Add to map
-      Polyline polyline = map.addPolyline(options);
-      polyline.setTag(mapPolygon);
-
-      // Style polyline
-      polyline.setStartCap(new RoundCap());
-      polyline.setEndCap(new RoundCap());
-      polyline.setWidth(getPolylineStrokeWidth());
-      polyline.setColor(parseColor(mapPolygon.getStyle().getColor()));
-      polyline.setJointType(JointType.ROUND);
-
-      polylines.add(polyline);
+    PolylineOptions options = new PolylineOptions();
+    options.clickable(false);
+    ImmutableList<LatLng> vertices = stream(mapPolygon.getVertices())
+        .map(GoogleMapsMapAdapter::toLatLng)
+        .collect(toImmutableList());
+    // Add vertices to PolylineOptions
+    options.addAll(vertices);
+    // Add to map
+    Polyline polyline = map.addPolyline(options);
+    polyline.setTag(mapPolygon);
+    // Style polyline
+    if (!isPolygonCompleted(mapPolygon.getVertices())) {
+      BitmapDescriptor customCap = bitmapDescriptorFromVector();
+      polyline.setStartCap(new CustomCap(customCap));
+      polyline.setEndCap(new CustomCap(customCap));
     }
+    polyline.setWidth(getPolylineStrokeWidth());
+    polyline.setColor(parseColor(mapPolygon.getStyle().getColor()));
+    polyline.setJointType(JointType.ROUND);
+
+    polylines.add(polyline);
+    polygons.put(mapPolygon, vertices);
+  }
+
+  private boolean isPolygonCompleted(List<Point> vertices) {
+    return vertices.get(vertices.size() - 1) == vertices.get(0);
+  }
+
+  private BitmapDescriptor bitmapDescriptorFromVector() {
```

Comment:
See comment in previous review about refactoring part of this into BitmapUtil.

Reply:
moved to BitmapUtil.